### ISSUE TITLE
Exam: Discrepancy in Student Review

### ISSUE THREAD

**Issue by mitcoef** [ls1intum/Artemis#7124]
### Describe the bug

Students can see a mixture of first and second assessment results when reviewing text exercises in an exam. 

### To Reproduce

1. Create an exam containing a text exercise and set number of correction rounds to 2
2. Register a student and conduct the exam 
3. As the student, complete the exam
4. As the instructor, assess the text exercise, making sure to change points and feedback items in the second correction round
6. As the student, review the exam

### Expected behavior

The student should only be able to see the results of the second correction round, not the first.

### Screenshots

![image](https://github.com/ls1intum/Artemis/assets/36709930/2b87ddeb-dca1-46f5-a9d2-bbbe914d5105)
![image](https://github.com/ls1intum/Artemis/assets/36709930/408ec97a-2214-4ce7-83d5-3bbe7d55757f)

### Which version of Artemis are you seeing the problem on?

6.4.1

### What browsers are you seeing the problem on?

Firefox

### Additional context

When reviewing complaints, instructors sometimes see results of the first assessment instead of the second. We were unable to reproduce this issue.

**Comment by mitcoef** [comment]
Instead of duplicate image, the second image is supposed to be:
![image](https://github.com/ls1intum/Artemis/assets/36709930/408ec97a-2214-4ce7-83d5-3bbe7d55757f)

**Comment by jerhard** [comment]
>The student should only be able to see the results of the second correction round, not the first.

Additionally, it is preferable if students only receive the JSON-data for the second review (not the first) from the server, so that they cannot access the result of the first review in case they inspect the transmitted JSON-data.

**Comment by tobias-lippert** [comment]
The current behavior is consistent with e.g., TUMExam. In the exam review you also see every feedback of each correction round.

So I don't think that's a bug.

**Comment by jerhard** [comment]
No, I do not think so. As you can see above, the percentage is inconsistent with the given points (60% vs. 10 / 10). Our understanding so far is that only the second or final correction should be displayed to the student.

**Comment by mitcoef** [comment]
Additionally, even if it was beneficial to display results of the first correction round (I strongly believe this is not the case), it should at least be obvious. As it is, there's just two unrelated scores. How do you even know which of these is the actual result and which is just the result of the first assessment?  
To be clear, the final result in this example is 6/10 points, not 10/10...

Consider the following scenario:
A student is missing a point to pass the exam and is thus looking for mistakes in the assessment during review.
There is an actual mistake in the assessment for this particular exercise, but the student just sees 10/10 points and does not further consider the exercise. They find no additional points and unfairly fail the exam. 

TLDR:
The final score, at the very least, needs to be clearly marked as such, leaving no room for ambiguity.

**Comment by tobias-lippert** [comment]
I'm not sure if I understand the issue correctly. So, the feedbacks and points are both from the second correction round?

**Comment by mitcoef** [comment]
The issue is that, during review as a student, you can simultaneously see 10/10 and 6/10 points with no indication which of the two scores is the *actual* score.

**Comment by Strohgelaender** [comment]
Can you confirm that the different scores (10 points and 6 points) are the results from correction round 1 and 2? Or do we have a similar case like in issue #7131 where the same result is shown, but feedback items were filtered out?

**Comment by mitcoef** [comment]
I just retested, giving 1 point in the first correction round and 10 (full marks) in the second. Result:
![image](https://github.com/ls1intum/Artemis/assets/36709930/c8c9a56f-3d05-46c7-9bb5-bbd5161962d6)

So yes, it's a mix of both.
